src/database/postgresql_handler.py

```
@@ -531,28 +531,56 @@ def commit(self) -> None:
         """
         if not self._connection:
             raise DatabaseError("Database not connected")
 
         try:
             if DRIVER == "pg8000":
-                # pg8000.native is in autocommit mode, no rollback possible
-                logger.warning("pg8000.native doesn't support rollback (autocommit mode)")
+                try:
+                    self._connection.run("ROLLBACK")
+                    logger.debug("pg8000 ROLLBACK sent")
+                except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)